### PR TITLE
Change language version to the latest major release (Lombiq Technologies: OCORE-104)

### LINKED ISSUE #10850: Update LangVersion to 9 at least, or rather 10
### Is your feature request related to a problem? Please describe.
Currently, the allowed C# language version depends on the development environment. Instead, we should specify it explicitly.

### Describe the solution you'd like
Specify `<LangVersion>10</LangVersion>` in `OrchardCore.Commons.props`.

### Describe alternatives you've considered
I guess this is not as trivial as this, since this config was changed around a lot: https://github.com/OrchardCMS/OrchardCore/pull/9017 configured it as 9.0, https://github.com/OrchardCMS/OrchardCore/pull/9079 reverted this to 8.0, and https://github.com/OrchardCMS/OrchardCore/pull/10038 removed the property altogether. Maybe we need to drop .NET Core 3.1 support for this?

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #10850

### CODE REVIEW

File: src/OrchardCore.Build/OrchardCore.Commons.props [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <Import Project="Dependencies.props" />
 
   <PropertyGroup>
+    <LangVersion>latestMajor</LangVersion>
```
Comment (Piedone) [inline src/OrchardCore.Build/OrchardCore.Commons.props:6]: I'd use a specific version, i.e. `10.0`, so it doesn't depend on the SDK you have installed, and we can update it deliberately.

**Review by Piedone — CHANGES_REQUESTED** [review]
It's really as simple as this? Nothing breaks, not even Razor IntelliSense.

File: src/OrchardCore.Build/OrchardCore.Commons.props (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Build/OrchardCore.Commons.props:6]: At the beginning I was planned to set it to `10.0` but I prefered to use `latestMajor` to make sure that we no longer need to update it

> so it doesn't depend on the SDK you have installed

Right, but while each OC version required specific framework it fine IMHO

Comment (Piedone) [reply src/OrchardCore.Build/OrchardCore.Commons.props:6]: We've multi-targeted frameworks before and most possibly we'll continue to always target the latest version and the latest LTS one, so the framework alone is not a decisive factor.

Comment (hishamco) [reply src/OrchardCore.Build/OrchardCore.Commons.props:6]: No problem to change it to `10.0`

**Comment by hishamco** [comment]
> It's really as simple as this? Nothing breaks, not even Razor IntelliSense.

I just run the SAAS app in my local without an issue, let me verify from Razor IntelliSense too, may be the issue on the past the multi-targeting .. may be

**Review by sebastienros — APPROVED** [review]
Merge when it's ready, I don't mind about both options

**Comment by hishamco** [comment]
@Piedone I can verify the IntelliSense works fine, so can go with `latestMajor` or shall we specify the version implicitly?

**Comment by Piedone** [comment]
Great then! Since nothing new came up since my [comment here](https://github.com/OrchardCMS/OrchardCore/pull/12452#discussion_r976399643), I still am on the opinion that we should explicitly specify the language version.

**Review by Piedone — APPROVED** [review]

**Pull request merged**